### ISSUE TITLE
consul.query_service constrains access to service data

### ISSUE THREAD

**Issue by nzlosh** [StackStorm-Exchange/stackstorm-consul#4]
Hello,

First of all, thank you for this great pack!

I have a question around the `query_service` action.  As it currently functions, the action returns a list of addresses which make up a service.

To demonstrate: here is the current returned data for this action:
```
10.189.1.2,10.189.0.36
```
Here is the same query without iterating over the address fields:
```
{
  "index": "9675425",
  "service": [
    {
      "Node": "aiur.dev.dc3",
      "TaggedAddresses": null,
      "ModifyIndex": 9675420,
      "ServicePort": 0,
      "ServiceName": "automation",
      "ServiceAddress": "aiur.dev.dc3.example.com",
      "ServiceID": "automation",
      "Address": "10.189.1.2",
      "ServiceTags": [
        "dev",
        "tag1",
        "tag2"
      ],
      "CreateIndex": 9674977,
      "ServiceEnableTagOverride": false
    },
    {
      "Node": "carlos.dev.dc3",
      "TaggedAddresses": null,
      "ModifyIndex": 9674920,
      "ServicePort": 0,
      "ServiceName": "automation",
      "ServiceAddress": "carlos.dev.dc3.example.com",
      "ServiceID": "automation",
      "Address": "10.189.0.36",
      "ServiceTags": [
        "dev",
        "tag1"
      ],
      "CreateIndex": 9674920,
      "ServiceEnableTagOverride": false
    }
  ]
}
```

My use case for this action is to fetch all member nodes in a service filtered by particular tags and return their `ServiceAddress`.  As seen above, there is a lot more data available than node addresses for a service.  Could someone explain why access to this data is constrained and if a pull request would be accepted to change this behaviour?

Relating to the same use case, it's not currently possible to supply multiple tags because the field type is `string`.  Would it be possible to change this to a list type?

**Comment by LindsayHill** [comment]
@nzlosh I don't think there's any specific reason why access to that data is constrained - probably because it met the needs of the original author. Pull Requests are always welcome - they often provoke good discussion.

Seems reasonable to change that tag type to list too. 

Might be worth pinging @TuxInvader here - I know he was doing some work with ST2 + Consul a few months ago. He might have some opinions on it.

**Comment by TuxInvader** [comment]
@nzlosh @LindsayHill I just modified the query_service to optionally return the ServicePort with the Address. I don't know what the original motivation was to just return the Address? But I don't see/use ServiceAddress fields in my environment, and I'm guessing the original author didn't either. 

Consul documentation suggest that ServiceAddress should be used if set, and node Address used as a fallback. See: https://www.consul.io/docs/agent/http/catalog.html#catalog_service

If modifying the action to return ServiceAddress when available and falling back to Address when not meets your requirement, then that might be the way to go.

WRT: The "tag" input, I also think changing to a list would be a good idea, but anyone using this action from a web-hook or chain might see some breakage? Doesn't st2 barf when it gets mismatched types? If so, then the SemVer spec will force us to 1.0.0 <https://www.youtube.com/watch?v=bW7Op86ox9g>

**Comment by nzlosh** [comment]
Thanks for your input @LindsayHill  and @TuxInvader.

I don't want to break backward compatibility just to fit my use case.  Would a pull request which contains a new action `consul.query_service_detailed`, which supports tags as a list and returns the full json output returned by consul be acceptable to you both as a solution?

**Comment by TuxInvader** [comment]
fine with me :-)

**Comment by LindsayHill** [comment]
Fixed by #8
